chocolatey/choco
# (#508) Change NuGet normal logging to be verbose logging

## Description Of Changes

This pull request updates the custom nuget logger we are using to not log normal informational output to the console unless `--verbose` is being used.
Additionally, this pull request updates the nuget logger, and the custom project context logging functionality to get rid of duplicated code.

## Motivation and Context

We do not want to obscure normal output for the user by including the information from NuGet.
While these can be helpful in some cases, hiding it behind a verbose flag by default make sense.

## Testing

1. Run `choco search 7zip.install`
2. Verify no output from NuGet is being shown
3. Run `choco search 7zip.install --verbose` (verbose in this case is synanymous with `--detailed`).
4. Verify the output from NuGet is shown with green text.
5. Run `choco info firefox`
6. Verify no output from NuGet is being shown
7. Run `choco info firefox --verbose`
8. Verify the output from NuGet is shown with green text.
9. Run `choco install trid --version 2.24.20221127`
10. Verify no output from NuGet is shown
11. Run `choco outdated`
12. Verify no output from NuGet is shown
13. Run `choco upgrade all`
14. Verify no output from NuGet is shown
14. Run `choco install 7zip.portable --version 19.0 --verbose`
15. Verify the output from NuGet is shown with green text.
16. Run `choco outdated --verbose`
17. Verify the output from NuGet is shown with green text.
18. Run `choco upgrade all --verbose`


### Operating Systems Testing

- Windows 11/10

## Change Types Made



## Change Checklist


## Related Issue

#508  
https://app.clickup.com/t/20540031/PROJ-395
